core/trino-spi/src/main/java/io/trino/spi/block/AbstractRowBlock.java

```diff
@@ -51,7 +51,7 @@ public final List<Block> getChildren()
     protected abstract boolean[] getRowIsNull();
 
     // the offset in each field block, it can also be viewed as the "entry-based" offset in the RowBlock
-    protected final int getFieldBlockOffset(int position)
+    public final int getFieldBlockOffset(int position)
```

Comment:
I think approach with external accessor like `ColumnarRow` might be better

Reply:
discussed offline. let's keep `getFieldBlockOffset` public as it's the simplest and most efficient approach.